abstract_domains/numerical/interval.py

```diff
@@ -0,0 +1,237 @@
+from abstract_domains.store import Store
+from abstract_domains.lattice import BottomMixin
+from abstract_domains.numerical.numerical import NumericalMixin
+from abstract_domains.state import State
+from core.expressions import *
+from typing import List, Set
+from math import inf
+
+from core.expressions_tools import ExpressionVisitor
+
+
+class Interval:
+    def __init__(self, lower=-inf, upper=inf):
+        """Create an Interval Lattice for a single variable.
+        """
+        assert lower <= upper
+        self._lower = None
+        self._upper = None
+        super().__init__()
+        self._lower = lower
+        self._upper = upper
+
+    @property
+    def interval(self):
+        return self.lower, self.upper
+
+    @interval.setter
+    def interval(self, bounds):
+        (lower, upper) = bounds
+        self.lower = lower
+        self.upper = upper
+
+    def add(self, other):
+        self.interval = (self.lower + other.lower, self.upper + other.upper)
+        return self
+
+    def sub(self, other):
+        self.interval = (self.lower - other.upper, self.upper - other.lower)
+        return self
+
+    def mult(self, other):
+        comb = [self.lower * other.lower, self.lower * other.upper, self.upper * other.lower, self.upper * other.upper]
+        self.interval = (min(comb), max(comb))
+        return self
+
+    def negate(self):
+        self.interval = (min(-self.lower, -self.upper), max(-self.lower, -self.upper))
+        return self
+
+    @property
+    def lower(self):
+        return self._lower
+
+    @lower.setter
+    def lower(self, b):
+        self._lower = b
+
+    @property
+    def upper(self):
+        return self._upper
+
+    @upper.setter
+    def upper(self, b):
+        self._upper = b
+
+    def __repr__(self):
+        return f"[{self.lower},{self.upper}]"
+
+
+class IntervalLattice(Interval, BottomMixin):
```

Comment:
This is a problem, yes. Should you even be able to calculate with a bottom interval? then the subclassing of interval is wrong and it should be aggregation